### DISCUSSION TITLE
An operator() for Type?

### DISCUSSION THREAD

**Discussion by fbleibel-g** [halide/Halide#6330] [Ideas]
Hi,

I have recently come across code of the form:

```
Type input_type = input.type();

Expr a = cast(input_type, expr_a);
Expr b = cast(input_type.widen(), expr_b1) * cast(input_type.widen(), expr_b2);
... more of the above ...
```

I thought that it might be nice and concise to offer casting directly from the Type itself. Something like this would avoid repetition and keep the code more readable:

```
Type input_type = input.type();
Type wide = input_type.widen();

Expr a = input_type(expr_a);
Expr b = wide(expr_b1) * wide(expr_b1);
```

This doesn't address other variants of casting (saturating_cast, ...others?), but I thought that it might be nice to be able to express casting from a local Type in this manner.

**Comment by steven-johnson** [comment]
My first reaction is that this sacrifices readability too much; from a casual scan, `input_type(expr_a)` is easy to misread as indexing-into-a-function.
